src/main/java/com/example/producerconsumer/QueueConsumerImpl.java

```diff
@@ -25,11 +25,16 @@ public QueueConsumerImpl(String name, List<String> messagesOutputList, QueueElem
     }
     
     @Override
+    //kolejność metod powinna być odwrotna
     public void consume(String operateOn) {
+        //ten kawalek kodu powinien być wydzielony - SRP
+
         String message;
         if (operateOn.equals(MathGeneratorService.POISON_PILL)) {
             message = String.format("Poison pill swallowed by: %s", name);
             messagesOutputList.add(message);
+
+            //logger ?
```

Comment:
I had a dilemma, normally I would use log4j but then the output be something like:
INFO com.example.producerconsumer.QueueConsumerImpl - Poison pill swallowed by C0 
vs current Poison pill swallowed by C0. So I choose the 2nd option.

Reply:
ja tam wole logger :)